testsuite/features/step_definitions/salt_steps.rb

```diff
@@ -505,6 +505,10 @@
   if use_salt_bundle
     if slemicro_host?(host)
       node.run('transactional-update --continue -n pkg rm venv-salt-minion', check_errors: false)
+      # SLE Micro could have also installed salt-minion
+      _result, code = node.run('rpm -q salt-minion')
```

Comment:
(Doesn't make that much of a difference)

We can  store which packages and repositories need to be removed in 2 string variables (or lists and then join them), do this check before running `transactional-update` and `rm -Rf` to modify them accordingly and call each of these commands once.

Reply:
Thank you for the suggestions, Paolo. This refactor is indeed a good idea but something for after 5.0.